client/src/modules/user/AuthBox/components/PasswordChangeForm.jsx

```diff
@@ -0,0 +1,296 @@
+import React, { Fragment, PureComponent } from 'react';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import _some from 'lodash/some';
+
+import AuthInput from './AuthInput';
+import { changePassword } from 'modules/user/model/actions';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import {
+  makeAbortablePromise,
+  validatePasswordStructure
+} from 'common/utils/helpers';
+import PendingButton from 'common/components/PendingButton';
+import { IntlPropType } from 'common/constants/propTypes';
+import { ValidationException } from 'common/exceptions/ValidationException';
+
+class PasswordChangeForm extends PureComponent {
+  pendingPromise = null;
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      newConfirmPasswordValue: undefined,
+      higherLevelErrors: {
+        newConfirmPasswordValueError: '',
+        newPasswordError: ''
+      },
+      isFormValid: false,
+      isNewPasswordValid: false,
+      isPasswordConfirmValid: false,
+      isPasswordValid: false,
+      newPassword: '',
+      password: '',
+      passwordChanged: false,
+      pending: false,
+      changePasswordErrorId: ''
+    };
+  }
+
+  componentDidUpdate() {
+    this.isFormValid();
```

Comment:
The "is" prefix indicates that this should be pure and just return a boolean indicating if the form is valid or not. Please rename to `validateForm` or something to clearly indicate that this is an effectful function call.